homeassistant/components/vacuum/reproduce_state.py

```diff
@@ -0,0 +1,84 @@
+"""Reproduce an Vacuum state."""
+import asyncio
+import logging
+from typing import Iterable, Optional
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import Context, State
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import (
+    ATTR_FAN_SPEED,
+    DOMAIN,
+    SERVICE_RETURN_TO_BASE,
+    SERVICE_SET_FAN_SPEED,
+    SERVICE_START,
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_RETURNING,
```

Comment:
What about `STATE_PAUSED` and `STATE_IDLE` ? I don't think we need `STATE_ERROR` since the user can't place the vacuum in this state but they can stop and pause.

Reply:
Good catch... Actually this PR is not ready for merging since some tests fail. tox is bothered by not having the service `vacuum.turn_on`, which seems really weird to me. Maybe it has something to do with the two implementations of `_BaseVacuum`

Reply:
I will add the missing states, but wanted to share this PR as soon as possible to ask for help.

Reply:
@Santobert correct the older vacuum model only used `turn_on` and `turn_off` and maybe even `vacuum.toggle` ? the new state model has all the new states :) I think MQTT and ecovacs are still on the older impelmentation? not sure about roomba.

Reply:
Yea, but I think this doesn't really matters here since the tests doesn't work with real devices

Reply:
I've added the missing states